layers/meta-balena-jetson/recipes-bsp/tegra-binaries/tegra186-flash-dry_32.4.2.bb

```diff
@@ -222,6 +223,7 @@ do_compile[depends] += " \
     virtual/bootloader:do_deploy \
 "
 
+do_compile[depends] += "u-boot-tegra:do_deploy"
```

Comment:
Do I need this uboot line? This was left over from the original PR someone added. 

Can you help me understand what this does @acostach ? Thanks!

Reply:
sure, you notice it's not do_compile_<machine_name>[depends] but do_compile[depends], so it's meant to apply to every device type that uses this package. As the Astro yocto machine extends the jetson-tx2, then this line applies to it too.

What it does actually is to make sure that u-boot gets built and deployed before this tegra186-flash-dry recipe can use the compiled u-boot binary to sign it. On TX2, DTBs and u-boot are signed by nvidia host binaries and then written to raw partitions on the emmc. Signing in this case is performed by
https://github.com/balena-os/balena-jetson/blob/c2d2576bcad333fe3815186ef3d26ba123ef68d5/layers/meta-balena-jetson/recipes-bsp/tegra-binaries/tegra186-flash-dry_32.4.2.bb#L130
which gets in  command line
https://github.com/balena-os/balena-jetson/blob/c2d2576bcad333fe3815186ef3d26ba123ef68d5/layers/meta-balena-jetson/recipes-bsp/tegra-binaries/tegra186-flash-dry_32.4.2.bb#L136
Therefore it parses the xml, finds u-boot there and performs the actual signing on it:
https://github.com/balena-os/balena-jetson/blob/c2d2576bcad333fe3815186ef3d26ba123ef68d5/layers/meta-balena-jetson/recipes-bsp/tegra-binaries/files/resinOS-flash186.xml#L362
and after the u-boot binary is signed, it becomes:
https://github.com/balena-os/balena-jetson/blob/c2d2576bcad333fe3815186ef3d26ba123ef68d5/layers/meta-balena-jetson/recipes-bsp/tegra-binaries/files/partition_specification186.txt#L20

Reply:
Got it thanks!

Reply:
Do I need it in this review? Obviously the other boards are working fine without it?